src/protocols/http/hpr_roaming_utils.erl

```diff
@@ -0,0 +1,80 @@
+%%%-------------------------------------------------------------------
+%%% @author jonathanruttenberg
+%%% @copyright (C) 2022, <COMPANY>
+%%% @doc
+%%%
+%%% @end
+%%% Created : 17. Sep 2022 10:56 AM
+%%%-------------------------------------------------------------------
+-module(hpr_roaming_utils).
+-author("jonathanruttenberg").
+
+%% API
+-export([
+    hexstring/1,
+    binary_to_hexstring/1,
+    format_time/1,
+    uint32/1,
+    hexstring_to_binary/1,
+    hexstring_to_int/1,
+    hexstring/2,
+    hex_to_binary/1
+]).
+
+-spec binary_to_hex(binary()) -> binary().
+binary_to_hex(ID) ->
+    <<<<Y>> || <<X:4>> <= ID, Y <- integer_to_list(X, 16)>>.
+
+-spec binary_to_hexstring(number() | binary()) -> binary().
+binary_to_hexstring(ID) when erlang:is_number(ID) ->
+    binary_to_hexstring(<<ID:32/integer-unsigned>>);
+binary_to_hexstring(ID) ->
+    <<"0x", (binary_to_hex(ID))/binary>>.
+
+-spec hexstring(number()) -> binary().
+hexstring(Bin) when erlang:is_binary(Bin) ->
+    binary_to_hexstring(Bin);
+hexstring(Num) when erlang:is_number(Num) ->
+    Inter0 = erlang:integer_to_binary(Num, 16),
+    Inter1 = string:pad(Inter0, 6, leading, $0),
+    Inter = erlang:iolist_to_binary(Inter1),
```

Comment:
Maybe move prepend of `0x` to call to `iolist_to_binary` to avoid constructing another binary?

Reply:
Maybe when `is_number` call `hexstring(Num, 6)` instead of duplicating the code?

Reply:
`hexstring` can probably call `binary_to_hexstring` to avoid some duplicated code.